### PR TITLE
Angle refactor + tests!

### PR DESCRIPTION
**Pull request by Bernstern**
# Changelog

- Added angle class to replace using double interchangeably for degrees + radians
- Added tests ci job
- Started adding tests
- Fixed a small but with code that checked if the line between two poses crossed a circle / simplified the logic

### CODE REVIEW

**Comment by Mineinjava** [comment]
the KalmanFilter branch is out of date. Can you change the PR to merge into main?

**Comment by Astr0clad** [comment]
Hey there! Could you please separate this PR into multiple? Ideally, it should be broken into one for the bugfix (highest priority) and then split the refactor of the Angle stuff into another PR. The tests are awesome, they've been on my to-do list for a while. 

tl;dr: Unrelated bugfix/features should be separated please

**Comment by Mineinjava** [comment]
I'm going to start including some of this stuff in main incrementally. Pinging @Astr0clad

File: quail/src/main/java/com/mineinjava/quail/pathing/Path.java [hunk 1]
```diff
@@ -186,42 +260,67 @@ public int nearestPointIndex(Pose2d point, int minIndex) {
...
+    // If min index is < 0, update it to 0 and TODO: throw a warning log
+    minIndex = max(minIndex, 0);
+
+    Pose2d nearestPoint = null, p;
 
-    for (int i = minIndex; i < points.size(); i++) {
-      Pose2d p = points.get(i);
-      double distance = Math.sqrt(Math.pow(point.x - point.x, 2) + Math.pow(point.y - point.y, 2));
+    // TODO(Bernie, pre merge): Change to infinity
+    double minDistance = 10000, distance;
```
Comment (Mineinjava) [inline quail/src/main/java/com/mineinjava/quail/pathing/Path.java:280]: It doesn't matter because the min distance has to be lower than the first one

**Pull request closed without merging**